src/plugins/share/public/types.ts

```diff
@@ -71,8 +72,25 @@ export interface ShareContextMenuPanelItem
  * directly in the context menu. If the item is clicked, the `panel` is shown.
  * */
 export interface ShareMenuItem {
-  shareMenuItem: ShareContextMenuPanelItem;
-  panel: EuiContextMenuPanelDescriptor;
+  shareMenuItem?: ShareContextMenuPanelItem;
+  // needed for Canvas
+  panel?: EuiContextMenuPanelDescriptor;
+  label?: 'PDF' | 'CSV' | 'PNG';
+  reportType?: string;
+  requiresSavedState?: boolean;
+  helpText?: ReactElement;
+  copyURLButton?: { id: string; dataTestSubj: string; label: string };
+  generateReportButton?: ReactElement;
+  generateReport?: Function;
+  generateReportForPrinting?: Function;
+  theme?: ThemeServiceSetup;
+  downloadCSVLens?: Function;
+  renderLayoutOptionSwitch?: boolean;
+  layoutOption?: 'print';
+  absoluteUrl?: string;
+  generateCopyUrl?: URL;
+  renderCopyURLButton?: boolean;
+  showRadios?: boolean;
```

Comment:
Whether to show the radios should be determined by how many options there are to show. If N>1, then we show the radios. (This is something for later cleanup)